react/src/components/ServiceLauncherPageContent.tsx

```diff
@@ -726,7 +726,7 @@ const ServiceLauncherPageContent: React.FC<ServiceLauncherPageContentProps> = ({
                         name="serviceName"
                         rules={[
                           {
-                            pattern: /^(?=.{4,24}$)\w[\w.-]*\w$/,
+                            pattern: /^(?=.{4,24}$)[\w-]+$/,
```

Comment:
Let's change the pattern to /^(?=.{4,24}$)\w[\w-]*\w$/
(disallow strings that starts/ ends with hyphen-)